### PR TITLE
NFC: various clean ups to ConvertConv2DToImg2ColPass

### PR DESCRIPTION
**Pull request by antiagainst**
* Use getter methods when possible
* Give dimensions and symbols better names

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp [hunk 1]
```diff
@@ -20,94 +21,96 @@ namespace Flow {
...
-    // TODO(ataei): Support for batched version.
-    if (inputShapeType.getShape()[0] > 1) return failure();
+    if (!filterType.hasStaticShape() || !inputType.hasStaticShape())
+      return failure();
 
     // TODO(ataei) : Support padding & dilation.
     if (!llvm::all_of(convOp.getDilations(), [](APInt element) {
           return element.getSExtValue() == 1;
         }))
       return failure();
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp]: style nit: wrap braces on simple single-statement bodies of if.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp [hunk 2]
```diff
@@ -20,94 +21,99 @@ namespace Flow {
...
+    if (!filterType.hasStaticShape() || !inputType.hasStaticShape())
+      return failure();
 
-    // TODO(ataei): Support for batched version.
-    if (inputShapeType.getShape()[0] > 1) return failure();
+    // TODO: Support for batched version.
+    if (inputType.getShape()[0] > 1) return failure();
 
-    // TODO(ataei) : Support padding & dilation.
+    // TODO: Support padding & dilation.
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp:74]: let's remove "padding" from the TODO.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp [hunk 3]
```diff
@@ -169,38 +171,30 @@ class Conv2DImg2ColMatmulConversion
...
-    if (!filterTensorType.hasStaticShape() || !inputTensorType.hasStaticShape())
+    auto inputType = convOp.getInputs()[0].getType().cast<RankedTensorType>();
+    auto filterType = convOp.getInputs()[1].getType().cast<RankedTensorType>();
+    auto outputType = convOp.getOutputs()[0].getType().cast<RankedTensorType>();
+
+    if (!filterType.hasStaticShape() || !inputType.hasStaticShape())
       return failure();
 
-    // TODO(ataei) : Support padding & dilation.
+    // TODO: Support padding & dilation.
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp:188]: style nit: add braces on simple single-statement bodies of if condition or make it a single line.

also, let's remove "padding" from TODO

**Review by hanhanW — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp [hunk 4]
```diff
@@ -20,94 +21,96 @@ namespace Flow {
...
+    const int fw = filterShape[1];
+    const int ic = filterShape[2];
+
     auto loc = convOp.getLoc();
 
-    // col tensor shape (n, d1, d1, k1, k2, ci)
-    SmallVector<int64_t, 4> colTensorShape = {outputShape[0], outputShape[1],
-                                              outputShape[2], filterShape[0],
-                                              filterShape[1], filterShape[2]};
+    SmallVector<int64_t, 4> colTensorShape = {n, oh, ow, fh, fw, ic};
```
[the remark this answers is not in the document]
Comment (antiagainst) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp:97]: `getStaticLoopRanges` is not guaranteed to be stable as there is no rule w.r.t. how those loops are ordered. It's better to be clear here by explicitly listing the dim sizes queried from filter/output shapes (as that's part of the op semantic).

[the remark this answers is not in the document]
Comment (hanhanW) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2DToImg2Col.cpp:101]: I see, thanks!

**Review by hanhanW — APPROVED** [review]
nice cleanups!

**Pull request merged**